### DISCUSSION TITLE
Add option to reset/clear search when search terms are removed from the search box

### DISCUSSION THREAD

**Discussion by MaxMattiello** [microsoft-search/pnp-modern-search#3016] [Ideas]
I've a similari issue but i do not understand how to solve it.
I've the attache page where i configured 3 Search Box and one Search Result.
Each Search Box is configured as follow:

Send teh query to a new page: ON
Page URL: i send all request to the actual page
Method: Query string parameter
Parameter Name: each Search Box has it's own parameter (CustomerOrderNo for the firstone, SerialNo for the second, Crossreference for the third)
Use dynamic data source as default input: ON
Connect to source: Page environment
Page environment's properties: Query String
Query String: Query parameters
queryParameters's properties: for each Search Boc i put the relative parameter (CustomerOrderNo for the firstone, SerialNo for the second, Crossreference for the third)
When insert a value in one of these Search Box, page is refreshed and Search Results shows me the correct documents but when i remove the value on the Search box, page is not refreshed so Search Results shows me previous documents.

Can you help me to fix?
![ModernSearch](https://user-images.githubusercontent.com/81249220/117530551-09fb5980-afde-11eb-8745-fa6d8a1b8026.png)

**Comment by wobba** [comment]
Use the bug template when filling out. It's there for a reason to fill in version number etc. Leaving out information makes repro hard. That said, removing something in the search box does not trigger re-execution. Search fires on terms+click.

**Comment by MaxMattiello** [comment]
Ops sorry!
Ok, so how can i reset all Search box values? Is there a possibility to perform a search when Search Box are empty?

**Comment by wobba** [comment]
@MaxMattiello not without the web parts adding this change - or you adding a link which clears the current search which trigger search with a default query if set.

**Comment by wobba** [comment]
I can rename your issue to "Add option to reset/clear search when search terms are removed from the search box" and keep it as an enhancement if you want?

**Comment by MaxMattiello** [comment]
@wobba yes please, rename this issue as enhancement. thanks!

**Comment by kasperbolarsen** [comment]
implemented in https://github.com/microsoft-search/pnp-modern-search/pull/2369
